src/org/labkey/test/components/ui/grids/QueryGrid.java

```diff
@@ -353,6 +402,13 @@ public List<FilterStatusValue> getFilterStatusFilterValues()
         }
 
         final WebElement removeAllFilters = Locator.tagWithClass("a", "remove-all-filters").refindWhenNeeded(this);
+
+        // The panel header element which will contain the Save and Undo buttons.
+        public WebElement panelHeader()
+        {
+            return getComponentElement().findElement(By.xpath("preceding-sibling::div[contains(@class,'panel-heading')]"));
```

Comment:
Use our `Locator` class instead of the default `By`. We decorate `WebElements` with extra retry capabilities.
```suggestion
            return Locator.xpath("preceding-sibling::div[contains(@class,'panel-heading')]").findElement(this);
```

Reply:
Done.